Resources/Prototypes/Reagents/Consumable/Drink/alcohol.yml

```diff
@@ -906,6 +906,22 @@
         reagent: Ethanol
         amount: 0.15
 
+- type: reagent
+  id: Eggnog
+  name: reagent-name-eggnog
+  parent: BaseAlcohol
+  desc: reagent-desc-eggnog
+  physicalDesc: reagent-physical-desc-fizzy-and-creamy
+  flavor: eggnog
+  color: "#e6d6bc"
+  metamorphicSprite:
+    sprite: Objects/Consumable/Drinks/eggnogglass.rsi
+    state: icon_empty
+  metamorphicMaxFillLevels: 4
+  metamorphicFillBaseName: fill-
+  metamorphicChangeColor: true
```

Comment:
```suggestion
  metamorphicChangeColor: false
```
Should be false if you already colored the fill levels, otherwise it will colored in the reagent's color again.
![grafik](https://github.com/user-attachments/assets/bc03a910-243d-4c84-8a4e-a86980e4c06d)
Also the fill level sprites should only have the pixels that show the liquid inside and not contain a copy of the glass sprite. Have a look at how the other drinks do it.